internal/extproc/translator/openai_gcpanthropic.go

```
@@ -9,90 +9,648 @@
+							Data: base64Data,
+						}
+						resultContent = append(resultContent, anthropic.NewDocumentBlock(pdfSource))
+					} else if isSupportedImageMediaType(contentType) {
+						resultContent = append(resultContent, anthropic.NewImageBlockBase64(contentType, base64Data))
+					} else {
+						return nil, fmt.Errorf("invalid media_type for image '%s'", contentType)
+					}
+				} else if strings.HasSuffix(strings.ToLower(imageURL), ".pdf") {
+					resultContent = append(resultContent, anthropic.NewDocumentBlock(anthropic.URLPDFSourceParam{
+						URL: imageURL,
+					}))
+				} else {
+					resultContent = append(resultContent, anthropic.NewImageBlock(anthropic.URLImageSourceParam{
+						URL: imageURL,
+					}))
+				}
+			case contentPart.InputAudioContent != nil:
+				return nil, fmt.Errorf("input audio content not supported yet")
+			}
+		}
+		return resultContent, nil
+	case openai.StringOrArray:
+		switch val := v.Value.(type) {
+		case string:
+			if val == "" {
+				return nil, nil
+			}
+			return []anthropic.ContentBlockParamUnion{
+				anthropic.NewTextBlock(val),
+			}, nil
+		case []openai.ChatCompletionContentPartUserUnionParam:
+			return openAIToAnthropicContent(val)
+		default:
+			return nil, fmt.Errorf("unsupported StringOrArray value type: %T", val)
+		}
+	}
+	return nil, fmt.Errorf("unsupported OpenAI content type: %T", content)
+}
+
+func extractSystemOrDeveloperPromptFromSystem(msg openai.ChatCompletionSystemMessageParam) string {
+	switch v := msg.Content.Value.(type) {
+	case string:
+		return v
+	case []openai.ChatCompletionContentPartUserUnionParam:
+		// Concatenate all text parts for completeness
+		var sb strings.Builder
+		for _, part := range v {
+			if part.TextContent != nil {
+				sb.WriteString(part.TextContent.Text)
+			}
+		}
+		return sb.String()
+	case nil:
+		return ""
+	default:
+		// If msg.Content is a StringOrArray, unwrap it
+		if soarr, ok := msg.Content.Value.(openai.StringOrArray); ok {
```

can we add this as a switch case instead of default?

---

Sure

---

done

---

but if it hits default, do we maybe now want to return an error?